src/generic/GenericTypes.jl

```diff
@@ -12,14 +12,14 @@
 
 const PermID = ObjectIdDict()
 
-mutable struct PermGroup <: Nemo.Group
-   n::Int
+mutable struct PermGroup{T<:Integer} <: Nemo.Group
+   n::T
 
-   function PermGroup(n::Int, cached=true)
+   function PermGroup(n::T, cached=true) where T<:Integer
       if haskey(PermID, n)
          return PermID[n]::PermGroup
```

Comment:
If you change it to `::PermGroup{T}`, it will be type stable.

Reply:
Did You mean typeassert `PermID[n]::PermGroup{T}`?

Reply:
Yes.